Extension/src/Utility/Process/streams.ts

```diff
@@ -0,0 +1,566 @@
+/* --------------------------------------------------------------------------------------------
+ * Copyright (c) Microsoft Corporation. All Rights Reserved.
+ * See 'LICENSE' in the project root for license information.
+ * ------------------------------------------------------------------------------------------ */
+
+import { Duplex, Readable, Writable } from 'stream';
+import { TextDecoder, TextEncoder } from 'util';
+import { ManualPromise } from '../Async/manualPromise';
+import { returns } from '../Async/returns';
+import { Signal } from '../Async/signal';
+import { EventStatus } from '../Eventing/interfaces';
+import { finalize } from '../System/finalize';
+import { is } from '../System/guards';
+import { verbose } from '../Text/streams';
+
+/* eslint-disable no-constant-condition */
+
+/**
+ * An iterator/iterable wrapper to process a stream of lines.
+ */
+export class LineIterator implements AsyncIterable<string>, AsyncIterator<string> {
+    #current = 0;
+    constructor(private lineBuffer: ReadableLineStream, private initial: number, private stopExpression?: string | RegExp) {
+        this.#current = Math.max(initial, lineBuffer.head);
+    }
+
+    get current() {
+        return Math.max(this.#current, this.lineBuffer.head);
+    }
+
+    advance() {
+        this.#current = Math.max(this.current + 1, this.lineBuffer.head);
+        return this;
+    }
+
+    /** this is both an iterator and the iterable itself. */
+    [Symbol.asyncIterator](): AsyncIterator<string, string> {
+        return this;
+    }
+
+    /**
+     * splits the iterator into two, so that they can be advanced independently.
+     *
+     * @returns a new iterator that starts at the current position.
+     */
+    tee() {
+        return new LineIterator(this.lineBuffer, this.current, this.stopExpression);
+    }
+
+    /**
+     * Stops the iterator at the point where we find the matching line.
+     *
+     * @param expression the string or regex to stop at
+     * @returns
+     */
+    until(expression: string | RegExp) {
+        this.stopExpression = expression;
+        return this;
+    }
+
+    /** allows the iterator to continue */
+    resume() {
+        this.stopExpression = undefined;
+        return this;
+    }
+
+    /**

```

Comment:
This comment style is different from the one above -- is that intentional?

Reply:
no idea.